auction_keeper/main.py

```diff
@@ -131,171 +141,171 @@ def __init__(self, args: list, **kwargs):
 
                 # Convert enough sin in woe to have woe >= sump + joy
-                if woe < sump and self.cat is not None:
-                    flog_amount = Wad(0)
+                if woe < (sump + joy) and self.cat is not None:
                     for bite_event in self.cat.past_bite(self.web3.eth.blockNumber):  # TODO: cache ?
                         era = bite_event.era(self.web3)
+                        now = self.web3.eth.getBlock('latest')['timestamp']
                         sin = self.vow.sin_of(era)
-                        if sin > Wad(0):
+                        # If the bite hasn't already been flogged and has aged past the `wait`
+                        if sin > Rad(0) and era + wait <= now:
                             self.vow.flog(era).transact()
 
                             # flog() sin until woe is above sump + joy
-                            if self.vow.woe() - self.vow.joy() >= sump:
+                            joy = self.vat.dai(self.vow.address)
+                            woe = (self.vat.sin(self.vow.address) - self.vow.sin()) - self.vow.ash()
+                            if woe - joy >= sump:
                                 break
 
-                # use heal() for removing the remaining joy
-                joy = self.vow.joy()
-                if joy > Wad(0):
-                    self.logger.debug(f"healing joy={self.vow.joy()} woe={self.vow.woe()}")
+                # use heal() for reconciling the remaining joy
+                joy = self.vat.dai(self.vow.address)
+                woe = (self.vat.sin(self.vow.address) - self.vow.sin()) - self.vow.ash()
+                if Rad(0) < joy <= woe:
+                    self.logger.info(f"healing joy={joy} woe={woe}")
                     self.vow.heal(joy).transact()
+                    # heal() changes joy and woe (the balance of surplus and debt)
+                    joy = self.vat.dai(self.vow.address)
+                    woe = (self.vat.sin(self.vow.address) - self.vow.sin()) - self.vow.ash()
 
-                if woe < sump and self.cat is None:
-                    self.logger.warning('Not enough woe to flop() and Cat address is not known !')
-                else:
-                    # Start a flop auction
+                if sump <= woe and joy == Rad(0):
                     self.vow.flop().transact()
 
-            if woe + sin >= sump and dai_balance <= min_balance:
-                self.logger.warning('Flop auction is possible but not enought DAI balance available to participate')
-
     def check_all_auctions(self):
         for id in range(1, self.strategy.kicks() + 1):
-            self.check_auction(id)
+            with self.auctions_lock:
+                if self.check_auction(id):
+                    self.feed_model(id)
+
+    def check_for_bids(self):
+        with self.auctions_lock:
+            # self.logger.debug(f"Checking for bids in {len(self.auctions.auctions)} auctions")
```

Comment:
not a huge fan of having log statements commented out and pushing them. Lets either include it as debug or remove it